### PR TITLE
Update build image instructions for Gen 2

### PR DESCRIPTION
**Pull request by Jay2113**
#### Description of changes:

The Amazon Linux 2023 image will be used by default for new Gen 2 applications. Existing Gen 2 applications can opt in to use this image via an environment variable.

#### Related GitHub issue #, if available:

N/A

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries
- [x] amplify-gen2

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?
An image asset was removed and redirects aren't needed.

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/gen2/quickstart/deploy-and-host.mdx [hunk 1]
```diff
@@ -15,15 +15,11 @@ If you already have your project remotely hosted in a Git repository, you can sk
...
 
-4. Under **Advanced settings**, navigate to **Build image** and enter `amplify:al2023`. The Amplify Amazon Linux 2023 build image supports Node.js versions 18 and 20 at build time. You can specify the Node.js version during your build through [live package updates](https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html#setup-live-updates) or [nvm](https://github.com/nvm-sh/nvm). Amplify Hosting will automatically deploy your compute runtime to match the Node.js version used at build time.
+<Callout warning>If you have an existing Gen 2 application, you can modify the build image by navigating to **Hosting > Environment variables** and creating a new environment variable `_CUSTOM_IMAGE` with the value `amplify:al2023`, then redeploying the build.</Callout>
```
Comment (renebrandel) [inline src/fragments/gen2/quickstart/deploy-and-host.mdx:20]: IMO, let's cut this. Quick start should assume the customer is running through it the first time.

File: src/fragments/gen2/quickstart/deploy-and-host.mdx [hunk 2]
```diff
@@ -15,15 +15,11 @@ If you already have your project remotely hosted in a Git repository, you can sk
 
 3. In the Git provider screen, choose **Option 2: Start with an existing app**. Connect the repository you just deployed and pick a branch.
 
-<Callout warning>Next.js 14 apps require a [minimum node version](https://nextjs.org/blog/next-14#other-changes) of 18.17 at build time.</Callout>
+<Callout> For new Gen 2 applications, Amplify Amazon Linux 2023 build image will be used by default which supports Node.js versions 18 and 20 at build time. You can specify the Node.js version during your build through [live package updates](https://docs.aws.amazon.com/amplify/latest/userguide/custom-build-image.html#setup-live-updates) or [nvm](https://github.com/nvm-sh/nvm). Amplify Hosting will automatically deploy your compute runtime to match the Node.js version used at build time. </Callout>
```
[the remark this answers is not in the document]
Comment (swaminator) [reply src/fragments/gen2/quickstart/deploy-and-host.mdx:18]: Yes lets kill it.

File: src/fragments/gen2/quickstart/deploy-and-host.mdx (hunk 1 above)
Comment (swaminator) [reply src/fragments/gen2/quickstart/deploy-and-host.mdx:20]: Agree

File: src/fragments/gen2/quickstart/deploy-and-host.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jay2113) [reply src/fragments/gen2/quickstart/deploy-and-host.mdx]: Done!

File: src/fragments/gen2/quickstart/deploy-and-host.mdx (hunk 1 above)
Comment (Jay2113) [reply src/fragments/gen2/quickstart/deploy-and-host.mdx]: Done!

**Review by renebrandel — APPROVED** [review]

**Review by swaminator — APPROVED** [review]

**Review by aspittel — APPROVED** [review]

**Pull request merged**